code/modules/mob/living/carbon/species.dm

```diff
@@ -863,14 +865,17 @@ var/global/list/whitelisted_species = list("Human")
 	anim(target = H, a_icon = 'icons/mob/mob.dmi', flick_anim = "dust-g", sleeptime = 15)
 	var/mob/living/adamantine_dust/A = new(H.loc)
 	if(golemmind)
+		var/golem_key = H.mind.key
+		has_died_as_golem.Add(golem_key)
+		spawn(GOLEM_RESPAWN_TIME)
```

Comment:
Don't use spawn for this. Better would be to store the `world.time` they died in the list too, then see if `world.time > death_time + GOLEM_RESPAWN_TIME`